donkeycar/parts/tub_v2.py

```diff
@@ -120,13 +124,48 @@ def run(self, *args):
         self.tub.write_record(record)
         return self.tub.manifest.current_index
 
+    def run_threaded(self, *args):
+        assert len(self.tub.inputs) == len(args), \
+            f'Expected {len(self.tub.inputs)} inputs but received {len(args)}'
+        record = dict(zip(self.tub.inputs, args))
+        # push the record into queue
+        self.record_queue.put(record)
+        self.mutex.acquire()
+        count = self.tub.manifest.current_index
+        self.mutex.release()
+        return count
+
+    def update(self):
+        self.on = True
+        while self.on or not self.record_queue.empty() > 0:
+            try:
+                # pop a record
+                record = self.record_queue.get()
+                # write the record to disk
+                self.mutex.acquire()
+                self.tub.write_record(record)
+                self.mutex.release()
+                # notify this task is done
+                self.record_queue.task_done()
+            except Empty:
+                pass
```

Comment:
If we try/catch, then we should add a logger.error here, and not silently go over this. If the write fails, the tub might be corrupt.